SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetDriverRPC.h

```diff
@@ -0,0 +1,241 @@
+						  Worker_EntityId EntityId, Worker_ComponentId ComponentId, uint64 RPCId, const FSpatialGDKSpanId& SpanId);
+	static void OnDataWritten(TArray<FWorkerComponentData>& OutArray, Worker_EntityId EntityId, Worker_ComponentId ComponentId,
+							  Schema_ComponentData* InData);
+	static void OnUpdateWritten(TArray<UpdateToSend>& OutUpdates, Worker_EntityId EntityId, Worker_ComponentId ComponentId,
+								Schema_ComponentUpdate* InUpdate);
+
+	void FlushUpdates(TArray<UpdateToSend>&);
+	bool CanExtractRPC(Worker_EntityId);
+	bool CanExtractRPCOnServer(Worker_EntityId);
+	bool ApplyRPC(Worker_EntityId, SpatialGDK::ReceivedRPC, const FRPCMetaData& MetaData);
+
+	USpatialNetDriver* NetDriver;
+
+	USpatialLatencyTracer* LatencyTracer;
+	SpatialGDK::SpatialEventTracer* EventTracer;
+
+	TUniquePtr<SpatialGDK::RPCService> RPCService;
+	// TUniquePtr<SpatialGDK::TRPCBufferReceiver<FRPCPayload, TimestampAndETWrapper>> NetMulticastReceiver;
+};
+
+/**
+ * Server side of the RPC component.
+ * Contains client, cross server and multicast senders
+ * Contains server and cross server receivers
+ * Able to collect initial RPC data for entities to create
+ */
+UCLASS()
+class SPATIALGDK_API USpatialNetDriverServerRPC : public USpatialNetDriverRPC
+{
+	GENERATED_BODY()
+public:
+	USpatialNetDriverServerRPC();
+	void Init(USpatialNetDriver* InNetDriver, const SpatialGDK::FSubView& InActorAuthSubView,
+			  const SpatialGDK::FSubView& InActorNonAuthSubView) override;
+
+	void ProcessReceivedRPCs() override;
+	void FlushRPCUpdates() override;
+
+	TUniquePtr<SpatialGDK::TRPCQueue<FRPCPayload, FSpatialGDKSpanId>> ClientReliableQueue;
+	TUniquePtr<SpatialGDK::TRPCQueue<FRPCPayload, FSpatialGDKSpanId>> ClientUnreliableQueue;
+	TUniquePtr<SpatialGDK::TRPCQueue<FRPCPayload, FSpatialGDKSpanId>> NetMulticastQueue;
+
+protected:
+	void GetRPCComponentsOnEntityCreation(const Worker_EntityId EntityId, TArray<FWorkerComponentData>& OutData) override;
+
+	TUniquePtr<SpatialGDK::RPCBufferSender> ClientReliableSender;
+	TUniquePtr<SpatialGDK::RPCBufferSender> ClientUnreliableSender;
+	TUniquePtr<SpatialGDK::TRPCBufferReceiver<FRPCPayload, TimestampAndETWrapper>> ServerReliableReceiver;
+	TUniquePtr<SpatialGDK::TRPCBufferReceiver<FRPCPayload, TimestampAndETWrapper>> ServerUnreliableReceiver;
+};
+
+/**
+ * Client side of the RPC component.
+ * It contains server senders, and client receivers.
+ */
+UCLASS()
+class SPATIALGDK_API USpatialNetDriverClientRPC : public USpatialNetDriverRPC
+{
+	GENERATED_BODY()
+public:
+	USpatialNetDriverClientRPC();
+	void Init(USpatialNetDriver* InNetDriver, const SpatialGDK::FSubView& InActorAuthSubView,
+			  const SpatialGDK::FSubView& InActorNonAuthSubView) override;
+
+	void ProcessReceivedRPCs() override;
+	void FlushRPCUpdates() override;
+
+	TUniquePtr<SpatialGDK::TRPCQueue<FRPCPayload, FSpatialGDKSpanId>> ServerReliableQueue;
```

Comment:
I'd prefer  these queues be registered with USpatialNetDriverRPC so that the parent class can flush them all without require the children classes to do it.

Reply:
Problem with that is it would then constrain the queues type to be a single one. Unless we register a "FlushCallback" instead.